docs/column_transforms/new_cast.md

```diff
@@ -0,0 +1,36 @@
+
+
+# new_cast
+
+Cast selected columns to a new type
+
+
+## Parameters
+
+| Argument |   Type    |                                    Description                                     |
+| -------- | --------- | ---------------------------------------------------------------------------------- |
+| casts    | cast_dict | A dict where the keys are columns and the values are the new type to cast them to. |
```

Comment:
I'm not seeing the corresponding YAML for this, but this look like a new type as well.  (I do think both of these new types are a good idea to add.)